qiita_db/artifact.py

```diff
@@ -0,0 +1,598 @@
+        prep_template : qiita_db.metadata_template.PrepTemplate, optional
+            If the artifact is being uploaded by the user, the prep template
+            to which the artifact should be linked to. If not provided,
+            `parents` should be provided.
+        parents : iterable of qiita_db.artifact.Artifact, optional
+            The list of artifacts from which the new artifact has been
+            generated. If not provided, `prep_template` should be provided.
+        processing_parameters : qiita_db.software.Parameter, optional
+            The processing parameters used to generate the new artifact
+            from `parents`. It is required if `parents` is provided. It should
+            not be provided if `prep_template` is provided.
+        can_be_submitted_to_ebi : bool, optional
+            Whether the new artifact can be submitted to EBI or not. Default:
+            `False`.
+        can_be_submitted_to_vamps : bool, optional
+            Whether the new artifact can be submitted to VAMPS or not. Default:
+            `False`.
+
+        Returns
+        -------
+        qiita_db.artifact.Artifact
+            A new instance of Artifact
+
+        Raises
+        ------
+        QiitaDBArtifactCreationError
+            If `filepaths` is not provided
+            If both `parents` and `prep_template` are provided
+            If none of `parents` and `prep_template` are provided
+            If `parents` is provided but `processing_parameters` is not
+            If both `prep_template` and `processing_parameters` is provided
+            If not all the artifacts in `parents` belong to the same study
+
+        Notes
+        -----
+        The visibility of the artifact is set by default to `sandbox`
+        """
+        # We need at least one file
+        if not filepaths:
+            raise QiitaDBArtifactCreationError(
+                "at least one filepath is required.")
+
+        # Parents or prep template must be provided, but not both
+        if parents and prep_template:
+            raise QiitaDBArtifactCreationError(
+                "parents or prep_template should be provided but not both")
+        elif not (parents or prep_template):
+            raise QiitaDBArtifactCreationError(
+                "at least parents or prep_template must be provided")
+
+        # If parents is provided, processing parameters should also be provided
+        if parents and not processing_parameters:
+            raise QiitaDBArtifactCreationError(
+                "if parents is provided, processing_parameters should also be"
+                "provided.")
+
+        # If prep_template is provided, processing_parameters should not be
+        # provided
+        if prep_template and processing_parameters:
+            raise QiitaDBArtifactCreationError(
+                "if prep_template is provided, processing_parameters should "
+                "not be provided.")
+
+        if timestamp is None:
```

Comment:
I don't think so - I was thinking in the case that we process something outside and we upload it through the CLI, we probably want to put the correct timestamp. Or do you think it is better to just remove that parameter and always default to "datetime.now()" since that is the time the artifact gets added to the system?